tests/lib/rule-tester/rule-tester.js

```diff
@@ -2295,6 +2228,123 @@ describe("RuleTester", () => {
         });
     });
 
+    describe("deprecations", () => {
+        let spy;
+
+        beforeEach(() => {
+            spy = sinon.spy(process, "emitWarning");
+        });
+
+        afterEach(() => {
+            spy.restore();
+        });
+
+        it("should log a deprecation warning when using the legacy function-style API for rule", () => {
+
+            /**
+             * Legacy-format rule (a function instead of an object with `create` method).
+             * @param {RuleContext} context The ESLint rule context object.
+             * @returns {Object} Listeners.
+             */
+            function functionStyleRule(context) {
+                return {
+                    Program(node) {
+                        context.report({ node, message: "bad" });
+                    }
+                };
+            }
+
+            ruleTester.run("function-style-rule", functionStyleRule, {
+                valid: [],
+                invalid: [
+                    { code: "var foo = bar;", errors: 1 }
+                ]
+            });
+
+            assert.strictEqual(spy.callCount, 1, "calls `process.emitWarning()` once");
+            assert.strictEqual(
+                spy.getCall(0).args[0],
```

Comment:
Instead of using `spy.getCall(0).args[0]`, you should actually use `assert.deepEqual(spy.getCall(0).args, ...)` to ensure we test ALL the args and not just the first one.

Reply:
Done.